include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -167,43 +168,50 @@ class pairwise_alignment_algorithm : protected policies_t...
 
         this->reset_optimum(); // Reset the tracker for the new alignment computation.
 
-        thread_local score_matrix_single_column<int32_t> local_matrix{};
-        local_matrix.reset_matrix(std::forward<sequence1_t>(sequence1), std::forward<sequence2_t>(sequence2));
-        auto matrix_iter = local_matrix.begin();
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
+        thread_local coordinate_matrix local_coordinate_matrix{};
 
-        initialise_column(*matrix_iter, sequence2);
+        local_score_matrix.reset_matrix(sequence1, sequence2);
+        local_coordinate_matrix.reset_matrix(
+                column_index_type{static_cast<size_t>(std::ranges::distance(sequence1) + 1)},
+                row_index_type{static_cast<size_t>(std::ranges::distance(sequence2) + 1)});
```

Comment:
Yes, it won't have an effect on runtime since there is no allocation happening. It should also not have an effect in SIMD. I wasn't sure about the construction thing. But in the godbolt example I am not seeing how the compiler would optimise away the entire memory allocation at least for the score matrix. I can remove the thread_local here and think about a solution for the vectorised code.